### PR TITLE
Feature/serp search stats

### PR DESCRIPTION
**Pull request by brindy**
Reviewer: Mia
Asana: https://app.asana.com/0/414235014887631/413609835171504/f
CC:

**Description**:

Cancels search requests that don't include the mobile search parameters used for statistics when a search is performed from the in-page search on duckduckgo.com

**Steps to test this PR**:
1. Navigate to duckduckgo.com
2. Search from within the page
3. Will be difficult to check without putting a breakpoint in the code, but ensure that nothing is broken

###### Reviewer Checklist:
- [x] **Ensure the PR solves the problem**
- [x] **Review every line of code**
- [x] **Ensure the PR does no harm by testing the changes thoroughly**
- [x] **Get help if you're uncomfortable with any of the above!**
- [x] Determine if there are any quick wins that improve the implementation

###### PR DRI Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications (Database connections, Grafana stats, CPU)

### CODE REVIEW

File: Core/WebViewController.swift [hunk 1]
```diff
@@ -157,19 +159,35 @@ open class WebViewController: UIViewController, WKNavigationDelegate, WKUIDelega
             decisionHandler(.allow)
             return
         }
-        
+
+        if (shouldReissueSearch(for: url)) {
```
Comment (subsymbolic) [inline Core/WebViewController.swift]: Remove the if statement extra brackets?

File: Core/AppUrls.swift [hunk 2]
```diff
@@ -72,7 +72,8 @@ public struct AppUrls {
     }
     
     public func isDuckDuckGo(url: URL) -> Bool {
-        return url.absoluteString.contains(Url.base)
+        guard let host = url.host else { return false }
+        return host == Url.base || host.hasSuffix(".\(Url.base)")
```
Comment (subsymbolic) [inline Core/AppUrls.swift]: Nice cleanup!

File: DuckDuckGoTests/AppUrlsTests.swift [hunk 3]
```diff
@@ -23,6 +23,76 @@ import XCTest
 class AppUrlsTests: XCTestCase {
     
     var mockStatisticsStore = MockStatisticsStore()
```
Comment (subsymbolic) [inline DuckDuckGoTests/AppUrlsTests.swift]: You never use this instantiation because you insantiate again in setup. Choose one or the other.

Comment (brindy) [reply DuckDuckGoTests/AppUrlsTests.swift]: thanks! I meant to change that

File: Core/WebViewController.swift (hunk 1 above)
Comment (brindy) [reply Core/WebViewController.swift]: old habits ...

**Pull request merged**